### ISSUE TITLE
My vision for the project's future

### ISSUE THREAD

**Issue by momobobe** [DUpdateSystem/UpgradeAll#175]
New here from F-Droid and glad to know this great project. I feel surprisingly odd that this App is so understated since I can tell your ambition would have many Android power users' dream come true. Although latest builds are still buggy, to be honest, I can see the project's potential and hope to make some useful contribution to its design in future (excuse me as I don't know much about coding and hence couldn't do coding stuff).

A typical use case for the project is, I have some Apps only available on GitHub or somewhere not a FOSS store (no Google Play or so as a serious geek always prefers FOSS solutions). To update them, in the past, I have to check manually one by one, or if smart enough, maintain a server and write some scripts to grab the updates. Either is quite troublesome for end users. F-Droid and Izzy's repo are good, but quite a lot of Apps are left out for various reasons. The ideal solution is to get everything done on my device solely with connection to the original download links for Apps, without any extra intermediate. And I wanna decide which Apps to download, no matter whether they're installed. This is the basic function I'd expect from the project, which is, however, unstable yet.

Furthermore, I'd like the project to be some kinda AIO solution for App distribution, which means anyone can publish their Apps here in a distributed manner, and anyone can discover indie Apps more easily. To make it, the App should be further decentralized with restricted reliance on remote servers, and with ability to easily customize, publish & share App lists. To ensure somewhat security and privacy, scanners for malware, ad, tracker, non-free dependency, excessive permission etc. should be a built-in feature, like https://github.com/DUpdateSystem/UpgradeAll/issues/82 mentioned. If possible, a distributed scoring and commenting system must be cool. All in all, I'm expecting a fully-functioned App store, distributed and FOSS though.

For your information, the ideas are partially implemented in several separated projects, for example, [Skydroid](https://skydroid.app/) is kinda decentralized, but unfortunately not very active now; [Aurora Store](https://gitlab.com/AuroraOSS/AuroraStore) has [εxodus scanner](https://reports.exodus-privacy.eu.org/en/) built in, and so as [Izzy's repo](https://android.izzysoft.de/repo/info) [[source](https://gitlab.com/IzzyOnDroid/repo/)] (with some of his own scanners in addition) (also many other Apps like [AppManager](https://github.com/MuntashirAkon/AppManager), [ClassyShark3xodus](https://directory.fsf.org/wiki/ClassyShark3xodus), [Warden](https://gitlab.com/AuroraOSS/AppWarden), etc. have that built in; besides, [LibChecker](https://github.com/zhaobozhen/LibChecker) provides a similar feature to scan libs); [Manyverse](https://www.manyver.se/) and [Secure Scuttlebutt protocol](https://github.com/ssbc) could be the base for the distributed implementation of scoring and commenting system (or even social network stuff if you wanna try).

So much for all these. I know this project is originally for Chinese users and lacking manpower now, however, for popularity and hence better development, I strongly suggest that it should get more exposure to international communities. Now very few people know your existence, although I have promoted a little bit (https://github.com/nbats/FMHYedit/issues/140, https://github.com/offa/android-foss/issues/231, https://github.com/pcqpcq/open-source-android-apps/issues/370, https://github.com/albertomosconi/foss-apps/issues/66), which is yet far from sufficient, and your team could have a look at some Android communities like those on Reddit for interaction with international audience. Moreover, some similar (but not as ambitious) projects like [ApkTrack](https://github.com/JusticeRage/ApkTrack), [APKUpdater](https://github.com/rumboalla/apkupdater) and [APKGrabber](https://github.com/hemker/apkgrabber) are slowly dying after few years of inactivity, and sincerely I wish your project a long life; also it could be a good idea to add support for APKMirror and Aptoide so that some of their users might get attracted here.

Finally, the most important task for the present is to debug and optimize what you have done so far, otherwise new users would lose interest when suffering poor network connection or lag issues.

**Comment by xz-dev** [comment]
Yeah, we hope this project can glue the various update channels together like glue. Via that to let more user to develop it.

**Comment by xz-dev** [comment]
The other important thing is we decide to move some function back to client such as get data from google play or other web site
